src/lib/ilp/index.js

```diff
@@ -194,7 +194,57 @@ class Ilp {
     _sha256 (preimage) {
         return Crypto.createHash('sha256').update(preimage).digest('base64');
     }
+
+    /**
+     * Decodes an Ilp Packet
+     *
+     * @returns {object} - Ilp packet as JSON object
+     */
+    decodeIlpPacket (inputIlpPacket) {
+        const binaryPacket = Buffer.from(inputIlpPacket, 'base64');
+        const jsonPacket = ilpPacket.deserializeIlpPayment(binaryPacket);
+        return jsonPacket;
+    }
+    
+    /**
+     * Get the transaction object in the data field of an Ilp packet
+     *
+     * @returns {object} - Transaction Object
+     */
+    getTransactionObject (inputIlpPacket) {
+        const jsonPacket = this.decodeIlpPacket(inputIlpPacket);
+        const decodedData = base64url.decode(jsonPacket.data.toString());
+        return JSON.parse(decodedData);
+    }
+    
+    /**
+     * Validate the transfer request against the decoded Ilp packet in it
+     *
+     * @returns {boolean} - True if the content in the transaction request is valid for Ilp packet
+     */
+    validateIlpAgainstTransferRequest (transferRequestBody) {
+        const transactionObject = this.getTransactionObject(transferRequestBody.ilpPacket);
+    
+        if (transferRequestBody.transferId !== transactionObject.transactionId) {
```

Comment:
Isn't this assuming that the 'transferId' in a POST /transfers message is same as the 'transactionId' used in POST /quotes?

Reply:
Hi @elnyry , Yes, you are correct. This model was agreed, I believe by the CCB, some two years ago as a method of linking quotes to transfers on a small key rather than the much larger, and expensive to index and search condition field. Maybe @MichaelJBRichards can confirm the status with the CCB.

Reply:
Thanks @bushjames - this has been proposed to the CCB, but has not been accepted (yet, there is a minor tweak done, which seems promising but it still will not apply for v1.x). We may have that change for v2.x release..